ag-grid-webcomponent/cell_editor/DecimalEditor.js

```diff
@@ -0,0 +1,174 @@
+            event.target.value.length +
+            this.eInput.selectionStart
+        );
+
+        this.eInput.value = newValue;
+        this.eInput.setSelectionRange(newSelectionIndex, newSelectionIndex);
+        return;
+      }
+
+      // Ignore formatting with invalid value
+      // We don't wanna block the user input
+      this.eInput.value = event.target.value;
+    });
+  }
+
+  afterGuiAttached() {
+    this.eInput.focus();
+  }
+
+  destroy() {}
+
+  getGui() {
+    return this.eInput;
+  }
+
+  getValue() {
+    const number = Number(this.localeParseFloat(this.eInput.value));
+    if (isNaN(number)) return null;
+
+    let formattedNumber = number.toFixed(this.decimalPlaces);
+
+    switch (this.type) {
+      case "number":
+        formattedNumber = Number(formattedNumber);
+        break;
+
+      default:
+        break;
+    }
+
+    return formattedNumber;
+  }
+
+  isCancelAfterEnd() {}
+
+  isPopup() {
+    return false;
+  }
+
+  // -- Utility functions -- //
+
+  countDecimals(value) {
+    if (Math.floor(value) === value) return 0;
+    let decimals = value.toString().split(this.decimalSeparator)[1];
+    return decimals?.length || 0;
+  }
+
+  cursorInLastPosition() {
+    return this.eInput.selectionStart == this.eInput.value.length;
+  }
+
+  escapeSeparator(value) {
+    return value.replace(/[.]/g, "\\$&");
+  }
+
+  formattableInput(event) {
+    if (!event.target.value) return false;
+
+    if (
+      event.data &&
+      event.data === "0" &&
+      this.cursorInLastPosition() &&
+      this.countDecimals(event.target.value) > 0
+    ) {
+      // Prevent formatting when inputting a zero as finishing decimal
+      return false;
+    }
+
+    const fixablePattern = new RegExp(
+      `^[-]?([${this.escapedThousandsSep}\\d]*)(${this.escapedDecimalSep}[0-9]{1,})?$`
+    );
+
+    return fixablePattern.test(event.target.value);
+  }
+
+  formatValue(value) {
+    if (!value) return null;
+
+    return new Intl.NumberFormat(this.countryCode, {
+      style: "decimal",
+      maximumFractionDigits: this.decimalPlaces,
+    }).format(value);
+  }
+
+  invalidDecimalInput() {
+    // Disallow adding more values than decimal places are supported
+
+    return (
+      this.cursorInLastPosition() &&
+      this.eInput.value.indexOf(this.decimalSeparator) >= 0 &&
+      this.countDecimals(this.eInput.value) >= this.decimalPlaces
+    );
+  }
+
+  invalidDecimalSeparatorInput() {
+    // Disallow adding another decimal separator
+
+    return (
+      this.decimalPlaces == 0 ||
+      this.eInput.value.indexOf(this.decimalSeparator) >= 0
+    );
+  }
+
+  isInputAllowed(char) {
+    if (this.invalidDecimalInput()) {
+      return false;
+    }
+
+    if (char == this.decimalSeparator && this.invalidDecimalSeparatorInput()) {
+      return false;
+    }
+
+    return /[0-9\.,-]/.test(char);
+  }
+
+  isKeyPressedNavigation(event) {
+    return event.key === "ArrowLeft" || event.key === "ArrowRight";
```

Comment:
Actually, this wasn't even ever checked because we've been listening to the `keypress` event which is only fired when a key is pressed down that produces a character value. But since that event is marked as deprecated anyway, I'm changing this to `keydown` and I will update the keys accordingly.